module/data/actor/templates/common.mjs

```diff
@@ -0,0 +1,67 @@
+import SystemDataModel from "../../abstract.mjs";
+import { MappingField } from "../../fields.mjs";
+import CurrencyTemplate from "../../shared/currency.mjs";
+import AbilityData from "../ability.mjs";
+
+/**
+ * A template for all actors that share the common template.
+ *
+ * @property {Object<string, AbilityData>} abilities  Actor's abilities.
+ * @mixin
+ */
+export default class CommonTemplate extends SystemDataModel.mixin(CurrencyTemplate) {
+
+  /** @inheritdoc */
+  static defineSchema() {
+    return this.mergeSchema(super.defineSchema(), {
+      abilities: new MappingField(new foundry.data.fields.EmbeddedDataField(AbilityData), {
+        initialKeys: CONFIG.DND5E.abilities
+      }, {label: "DND5E.Abilities"})
+    });
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritdoc */
+  static migrateData(source) {
+    this.migrateACData(source);
+    this.migrateMovementData(source);
```

Comment:
Can we make the internal migration functions private? Or do they need to be part of the public API here?

Reply:
Hard private isn't going to work for these methods because of how the mixin operates:

`TypeError: Failed data migration for NPCData: can't access private field or method: object is not the right class`

Reply:
It indeed does not work for instance methods, but it should work for static methods so long as we explicitly use the class name per this comment: https://github.com/foundryvtt/dnd5e/pull/1990#discussion_r1059464136